### PR TITLE
feat: add variables & feedback

### LINKED ISSUE #12: pjlink power state feedback
Is it possible to add a feedback that would change the background color depending on the power state of the projector?

### PR DESCRIPTION
**Pull request by jmcc94**
Closes #12 / #20 / #21

### CODE REVIEW

File: pjlink.js [hunk 1]
```diff
@@ -1,314 +1,617 @@
...
+					return true
+				}
+				break
+			case 'errorFilter':
+				if (self.errorFilter === feedback.options.errorState) {
+					return true
+				}
+				break
+			case 'errorOther':
+				if (self.errorFilter === feedback.options.errorState) {
```
Comment (josephdadams) [inline pjlink.js]: Should this be `self.errorOther`?

**Comment by josephdadams** [comment]
@jmcc94 left one comment. Otherwise looks good to me!

**Comment by istnv** [comment]
You may as well add feedback and variable for the shutter while you're working on it :)

File: pjlink.js (hunk 1 above)
Comment (jmcc94) [reply pjlink.js]: Yes it should! Fixed.

**Comment by jmcc94** [comment]
@istnv - Feedback/Variables added for both Freeze & Mute

File: HELP.md [hunk 2]
```diff
@@ -12,10 +12,20 @@
...
 
+- Error Status
+- Freeze Status
+- Input Horizontal Resolution
+- Input Vertical Resolution
+- Lamp Hour
+- Mute Status
 - Power Status
-- Projector Input
+- Projector Inputt
```
Comment (istnv) [inline HELP.md:22]: typo

**Comment by jmcc94** [comment]
Once you are happy, can this be merged? And then, the updated module go into beta..?

**Comment by istnv** [comment]
@jmcc94 I was going to work on this myself soon, which is why I suggested the other feedbacks.
You've done more than I had planned and it looks good.

**Comment by mabyoro** [comment]
wow that is a really great Upgrade, thanks a lot! I totally agree @istnv that this is more than I had planned and it looks good!

With my Panasonic-Projectors (RZ31, RZ770, DZ21, RZ120) it looks like Mute and input is reversed...
mute

**Comment by jmcc94** [comment]
> wow that is a really great Upgrade, thanks a lot! I totally agree @istnv that this is more than I had planned and it looks good!
> 
> With my Panasonic-Projectors (RZ31, RZ770, DZ21, RZ120) it looks like Mute and input is reversed... mute

fixed in #25

**Pull request merged**